### ISSUE TITLE
[v 1.1 (78) iPhone] Problems with Selling Eggs

### ISSUE THREAD

**Issue by lemoness** [HabitRPG/habitica-ios#224]
This may apply more broadly as well.

When I try to sell eggs, sometimes it highlights the egg and removes a different egg that I have. The website confirms that the correct eggs were sold, so this is a display issue.

![img_3656](https://cloud.githubusercontent.com/assets/3494081/10449704/1c617970-714a-11e5-8e55-a7b78111cba5.PNG)

**Comment by lemoness** [comment]
Possibly related: when using my Spooky Potions to hatch pets, they migrate elsewhere instead of disappearing. The website confirms that the Spooky Pets were successfully hatched.
![img_3657](https://cloud.githubusercontent.com/assets/3494081/10449764/73988846-714a-11e5-915f-075db427fb13.PNG)
![img_3658](https://cloud.githubusercontent.com/assets/3494081/10449763/739512a6-714a-11e5-9aaa-d7bfd2c5f5ee.PNG)

**Comment by vIiRuS** [comment]
Tried several things to reproduce this, but it always updated/removed the right row. Do you know when it happens? does it always happen or only sometimes/at random?

**Comment by lemoness** [comment]
So, it hasn't done it exactly like this again, but in trying to hatch some normal pets, I've had some additional weird behavior. Here's an example that I documented:

1. Decided to hatch a shade panda. When I selected it, the panda egg appeared randomly among the potions.
![img_3707](https://cloud.githubusercontent.com/assets/3494081/10555837/7d890848-742a-11e5-80eb-66cd6602c12a.jpg)

2. After successfully hatching a shade panda, the panda egg stayed highlighted, and the number remained the same, even after I went to hatch a fox:
![img_3708](https://cloud.githubusercontent.com/assets/3494081/10555846/a5c888f6-742a-11e5-9228-1a229f2e4c69.jpg)

3. Now both the fox and panda eggs, both highlighted, have appeared among the potions:
![img_3709](https://cloud.githubusercontent.com/assets/3494081/10555855/dd77b772-742a-11e5-8e0b-88f1003959b3.jpg)

4. And not only have they not disappeared, the panda egg has duplicated:
![img_3711](https://cloud.githubusercontent.com/assets/3494081/10555863/f39344f4-742a-11e5-8169-9fe3e9280c9f.jpg)

So while I wasn't able to reproduce the original bug, something odd is definitely still happening.
